src/main/java/chatbot/impl/MessageParserImpl.java

```diff
@@ -0,0 +1,215 @@
+package chatbot.impl;
+
+import java.util.List;
+
+import chatbot.MessageParser;
+import chatbot.Task;
+import chatbot.TaskStorage;
+import chatbot.exceptions.InvalidMessageException;
+import chatbot.impl.tasks.DeadlineTask;
+import chatbot.impl.tasks.EventTask;
+import chatbot.impl.tasks.TodoTask;
+
+/**
+ * Implements the MessageParser interface to handle various types of user messages.
+ */
+public class MessageParserImpl implements MessageParser {
```

Comment:
Would it be better to have named the interface MessageParserInterface (or similar) so that the class itself could just be named as MessageParser?